packages/taiko-client/driver/chain_syncer/blob/blocks_inserter/common.go

```
@@ -30,32 +35,68 @@ func createPayloadAndSetHead(
 		"parentHash", meta.Parent.Hash(),
 		"l1Origin", meta.L1Origin,
 	)
+	// Insert a TaikoL2.anchorV2 / TaikoAnchor.anchorV3 transaction at transactions list head,
+	// then encode the transactions list.
+	txListBytes, err := rlp.EncodeToBytes(append([]*types.Transaction{anchorTx}, meta.Txs...))
+	if err != nil {
+		log.Error("Encode txList error", "blockID", meta.BlockID, "error", err)
+		return nil, err
+	}
+
+	// If the Pacaya block is preconfirmed, we don't need to insert it again.
+	if meta.BlockID.Cmp(new(big.Int).SetUint64(rpc.PacayaClients.ForkHeight)) >= 0 {
+		header, err := isBlockPreconfirmed(ctx, rpc, meta, txListBytes, anchorTx)
+		if err != nil {
+			log.Debug("Failed to check if the block is preconfirmed", "error", err)
```

Shall we throw the error?

---

No need, if its not precomfired or its an invalid preconfirmed block, we insert that block or reorg the chain normally.